openapi/src/main/java/io/micronaut/openapi/visitor/ConvertUtils.java

```diff
@@ -102,6 +110,153 @@ public final class ConvertUtils {
     private ConvertUtils() {
     }
 
+    /**
+     * Convert the given Map to a JSON node and then to the specified type.
+     *
+     * @param <T> The output class type
+     * @param values The values
+     * @param context The visitor context
+     * @param type The class
+     *
+     * @return The converted instance
+     */
+    public static <T> T toValue(Map<CharSequence, Object> values, VisitorContext context, Class<T> type) {
+        JsonNode node = toJson(values, context);
+        try {
+            return ConvertUtils.treeToValue(node, type, context);
+        } catch (JsonProcessingException e) {
+            context.warn("Error converting  [" + node + "]: to " + type + ": " + e.getMessage(), null);
+        }
+        return null;
+    }
+
+    /**
+     * Convert the given map to a JSON node.
+     *
+     * @param values The values
+     * @param context The visitor context
+     *
+     * @return The node
+     */
+    public static JsonNode toJson(Map<CharSequence, Object> values, VisitorContext context) {
+        Map<CharSequence, Object> newValues = toValueMap(values, context);
+        return ConvertUtils.getJsonMapper().valueToTree(newValues);
+    }
+
+    public static Map<CharSequence, Object> toValueMap(Map<CharSequence, Object> values, VisitorContext context) {
+        Map<CharSequence, Object> newValues = new HashMap<>(values.size());
```

Comment:
`new HashMap<>(values.size())` is not correct as the map would have expand to have this capacity, please use `CollectionUtils::newHashMap`

Reply:
Yeah it’s v4